### PR TITLE
1362 editable post dates

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Adding post date
- Migration
- Test
- Validation

Test these changes by:
-

Fixes ushahidi/platform#1362

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1440)

### CODE REVIEW

File: src/Core/Entity/Post.php [hunk 1]
```diff
@@ -79,6 +80,7 @@ protected function getDefinition()
 			'status'          => 'string',
 			'created'         => 'int',
 			'updated'         => 'int',
+			'post_date'       => 'string',
```
Comment (rjmackay) [inline src/Core/Entity/Post.php]: We can use `*timestamp` here and it should run the value through `strtotime`

File: application/classes/Ushahidi/Repository/Post.php [hunk 2]
```diff
@@ -818,6 +818,10 @@ public function create(Entity $entity)
 
 		// Remove attribute values and tags
 		unset($post['values'], $post['tags'], $post['completed_stages'], $post['sets'], $post['source'], $post['color']);
+		// Convert post_date to mysql format
+		if(array_key_exists('post_date', $post)) {
+				$post['post_date'] = date("Y-m-d H:i:s", strtotime($post['post_date']));
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Post.php]: This should happen on the entity so its in a consistent format there

File: src/Core/Tool/Authorizer/PostAuthorizer.php [hunk 3]
```diff
@@ -61,7 +61,7 @@ class PostAuthorizer implements Authorizer, Permissionable
      */
     protected function getAllPrivs()
     {
-        return ['read', 'create', 'update', 'delete', 'search', 'change_status'];
+        return ['read', 'create', 'update', 'delete', 'search', 'change_status', 'change_timestamp'];
```
Comment (rjmackay) [inline src/Core/Tool/Authorizer/PostAuthorizer.php]: Don't need this change_timestamp isn't a separate permission, just part of `update`

Comment (willdoran) [inline src/Core/Tool/Authorizer/PostAuthorizer.php]: done

File: application/classes/Ushahidi/Repository/Post.php (hunk 2 above)
Comment (willdoran) [inline application/classes/Ushahidi/Repository/Post.php]: moved

**Pull request merged**